### PR TITLE
fix #308

### LINKED ISSUE #308: Users endpoint
I saw this ticket system and seems perfect for my needs.
The big problem is how to integrate my payment system to handesk when they pay.
Right now I have another helpdesk and they have endpoints to create users but this one miss this feature.
There are in case other ways to create users and disable the create accounts page?

### PR DESCRIPTION
**Pull request by Mte90**
This add 2 new endpoints:

* /api/users/create?name=prova2&email=ciao@ciao7.it&password=blablabla
* /api/users/

Require a bit of testing because I cannot find how to use the endpoints in handesk

### CODE REVIEW

File: routes/api.php [hunk 1]
```diff
@@ -4,7 +4,9 @@
     Route::resource('tickets', 'TicketsController', ['except' => 'destroy']);
     Route::post('tickets/{ticket}/comments', 'CommentsController@store');
     Route::post('tickets/{ticket}/assign', 'TicketAssignController@store');
+    Route::post('users/create', 'UsersController@store');
```
Comment (BadChoice) [inline routes/api.php]: actually the `create` is not required, a simple post to users will create them

File: app/Http/Controllers/Api/UsersController.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    	$users = User::take($num_results)->skip($offset)->get();
+    	return response()->json( $users );
+    }
+    
+    public function store(Request $request)
+    {
+      $users=new User;
+      $users->name=$request->get('name');
+      $users->email=$request->get('email');
+      $users->password=Hash::make($request->get('password'));
```
Comment (BadChoice) [inline app/Http/Controllers/Api/UsersController.php]: we usually use the bcrypt method `bcrypt($request('password'))`

Comment (Mte90) [reply app/Http/Controllers/Api/UsersController.php]: I saw that in the handesk code is used hash

File: routes/api.php (hunk 1 above)
Comment (Mte90) [reply routes/api.php]: Well an endpoint to create and one to list them, I can change the endpoint name in case

**Comment by Mte90** [comment]
So I don't need to add any tests for that?

File: app/Http/Controllers/Api/UsersController.php (hunk 2 above)
Comment (BadChoice) [reply app/Http/Controllers/Api/UsersController.php]: oh, ok then ;)

**Comment by BadChoice** [comment]
A test would be great!!
do you know how to add them?

File: routes/api.php (hunk 1 above)
Comment (BadChoice) [reply routes/api.php]: yes, `POST users` to create one, `GET users` list them

Comment (BadChoice) [reply routes/api.php]: `GET users/{user_id}` to the full info of one

**Pull request merged**